grpc/pb-marshalling.go

```diff
@@ -234,3 +237,118 @@ func argsToPerformValidationRequest(domain string, challenge core.Challenge, aut
 	}, nil
 
 }
+
+func registrationToPB(reg core.Registration) (*rapb.Registration, error) {
+	keyBytes, err := reg.Key.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+	ipBytes, err := reg.InitialIP.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	createdAt := reg.CreatedAt.UnixNano()
+	status := string(reg.Status)
+	var contacts []string
+	if reg.Contact != nil {
```

Comment:
It sounds like this aspect of the PR wasn't super intuitive, could you add a comment here describing some of what you saw with gRPC & the empty vs nil slice?